src/en/mehgazone/src/eu/kanade/tachiyomi/extension/en/mehgazone/Mehgazone.kt

```diff
@@ -0,0 +1,429 @@
+        val link: String,
+        val content: RenderedDto,
+        val excerpt: RenderedDto,
+    )
+
+    @Serializable
+    private data class RenderedDto(
+        val rendered: String,
+    )
+
+    private object DateTimeSerializer : KSerializer<Calendar> {
+        private val sdf by lazy { SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.US) }
+
+        override val descriptor = PrimitiveSerialDescriptor("DateTime", PrimitiveKind.STRING)
+        override fun serialize(encoder: Encoder, value: Calendar) = encoder.encodeString(sdf.format(value.time))
+        override fun deserialize(decoder: Decoder): Calendar {
+            val cal = Calendar.getInstance()
+            val date = sdf.parse(decoder.decodeString())
+            if (date != null) {
+                cal.time = date
+            }
+            return cal
+        }
+    }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    companion object {
+        private const val WORDPRESS_USERNAME_PREF_KEY = "WORDPRESS_USERNAME"
+        private const val WORDPRESS_USERNAME_PREF_TITLE = "WordPress username"
+        private const val WORDPRESS_USERNAME_PREF_SUMMARY = "The WordPress username"
+        private const val WORDPRESS_USERNAME_PREF_DIALOG = "To see your username:\n\n" +
+            "Go to https://bodysuit23.mehgazone.com/wp-admin/profile.php and you should see your username near the top of the page."
+        private const val WORDPRESS_USERNAME_PREF_DEFAULT_VALUE = ""
+
+        private const val WORDPRESS_APP_PASSWORD_PREF_KEY = "WORDPRESS_APP_PASSWORD"
+        private const val WORDPRESS_APP_PASSWORD_PREF_TITLE = "WordPress app password"
+        private const val WORDPRESS_APP_PASSWORD_PREF_SUMMARY = "The WordPress app password (not your account password)"
+        private const val WORDPRESS_APP_PASSWORD_PREF_DIALOG = "To setup:\n\n" +
+            "Go to https://bodysuit23.mehgazone.com/wp-admin/profile.php and you should be able to create a new app password near the bottom of the page."
+        private const val WORDPRESS_APP_PASSWORD_PREF_DEFAULT_VALUE = ""
+    }
+
+    private val authInterceptor: BasicAuthInterceptor by lazy {
+        BasicAuthInterceptor(
+            preferences.getString(WORDPRESS_USERNAME_PREF_KEY, WORDPRESS_USERNAME_PREF_DEFAULT_VALUE),
+            preferences.getString(WORDPRESS_APP_PASSWORD_PREF_KEY, WORDPRESS_APP_PASSWORD_PREF_DEFAULT_VALUE),
+        )
+    }
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        EditTextPreference(screen.context).apply {
+            val name = preferences.getString(WORDPRESS_USERNAME_PREF_KEY, WORDPRESS_USERNAME_PREF_DEFAULT_VALUE)!!
+
+            key = WORDPRESS_USERNAME_PREF_KEY
+            title = WORDPRESS_USERNAME_PREF_TITLE
+            dialogMessage = WORDPRESS_USERNAME_PREF_DIALOG.linkify()
+            summary = name.ifBlank { WORDPRESS_USERNAME_PREF_SUMMARY }
```

Comment:
```suggestion
            summary = "%s"
```
this should work, but sometimes doesn't so check

Reply:
when I use `"%s"` it just shows `%s` on the preferences screen